PHP/Toolset.md

```diff
@@ -8,45 +8,69 @@ Dependency management with Composer
 * Ways to autoload dependencies in Composer; 
 * composer.json vs composer.lock
 
+Live templates in IDE
+
+Error handling with [Whoops](https://github.com/filp/whoops)
 
 [Carbon](https://github.com/briannesbitt/Carbon): a simple PHP API extension for DateTime 
 
 Debugging app with [Kint](https://github.com/kint-php/kint)
 
 Editorconfig as a way to share common settings between diffeerent IDEs and editors
 
-# Middle Level
 Generation of universally unique identifiers with [uuid library](https://github.com/ramsey/uuid)
 
+Experience with one of the following template engines: Twig, Plates, Aura View etc
+
+# Middle Level
 Code complexity check with [PHPMD](https://github.com/phpmd/phpmd)
 
-Code debug with xDebug
+Code debug with xDebug. Simple and conditional breakpoints.
+
+Code profiling with xDebug 
+
+Code profiling with [XHProf](https://tideways.com/profiler/xhprof-for-php7)
 
 Fake data generation with [Faker](https://github.com/fzaninotto/Faker)
 
 PSR-7 HTTP message interfaces: [Zend Diactoros](https://github.com/zendframework/zend-diactoros)
 
-Experience with one of the following template engines: Twig, Plates, Aura View etc
-
-Error handling with [Whoops](https://github.com/filp/whoops)
+PSR-15: HTTP Server Request Handlers
 
 Experience with one of the following DB migration tools: [Phinx](https://github.com/cakephp/phinx), Doctrine Migrations etc
 
 Active Record ORM: Eloquent or similar
 
-Setting up VCS-hooks to run some tools (like pre-push hook for PHPUnit and PHPCS)
+Data Mapper ORM: Doctrine or similar
 
 Logging with [Monolog](https://github.com/Seldaek/monolog)
 
 Loading environment variables from .env file with [PHP dotenv](https://github.com/vlucas/phpdotenv)
 
 Experience in setting up a web-server: Apache or Nginx
+* Apache + mpd_php vs Nginx + php-fpm
 
 Running scripts with cron
 
+Developing composer libraries as a way to reuse code.
+
+Setting up VCS-hooks to run some tools (like pre-push hook for PHPUnit and PHPCS)
+
+Composer scripts, events, hooks
+
+Makefile as a way to share reusable scripts
 
+Working with queues using RabbitMQ, Kafka, Gearman, etc
+ 
+Experience with one of the following routers: Aura Router, Route (php league) etc
+ 
 # Senior Level
-Setting up a consistent development environment using Vagrant or Docker. 
+Setting up a consistent development environment using
+* Vagrant
+* Docker:
+    * Dockerfiles, pure and based on existing images
+    * docker-compose to join separate images into encapsulated network.
+    * optimizing images
```

Comment:
not sure that trimming an image to make it slimmer is a job for PHP developer

Reply:
Yeah, maybe I just got carried away :)